client/my-sites/post-type-list/post-actions-ellipsis-menu/duplicate.jsx

```diff
@@ -0,0 +1,72 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
+import { get } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import PopoverMenuItem from 'components/popover/menu-item';
+import QueryPostTypes from 'components/data/query-post-types';
+import { mc } from 'lib/analytics';
+import { canCurrentUser } from 'state/selectors';
+import { getPost } from 'state/posts/selectors';
+import { getPostType } from 'state/post-types/selectors';
+import { getCurrentUserId, isValidCapability } from 'state/current-user/selectors';
+import { getEditorDuplicatePostPath } from 'state/ui/editor/selectors';
+import config from 'config';
+
+function PostActionsEllipsisMenuDuplicate( { translate, siteId, canEdit, duplicateUrl, isKnownType } ) {
+	if ( ! config.isEnabled( 'posts/post-type-list' ) || ! canEdit ) {
+		return null;
+	}
+
+	function bumpStat() {
+		mc.bumpStat( 'calypso_cpt_actions', 'duplicate' );
+	}
+
+	return (
+		<PopoverMenuItem href={ duplicateUrl } onClick={ bumpStat } icon="pages">
+			{ siteId && ! isKnownType && <QueryPostTypes siteId={ siteId } /> }
```

Comment:
Typically, a Query component is placed as a sibling or otherwise close relative of the element that requires the data it requests (e.g. `<div><QueryPosts { ...props } /><Posts posts={ … } /></div>`), unless there's another specific reason (e.g. preloading). Could you provide some context behind its role here, so I can better understand?